### PR TITLE
Forms missing parameters

### LINKED ISSUE #300: [Forms] handle missing parameters when submitting forms
High priority:
- Set required fields as "required" in forms
- Display flash message instead of exception when a required field is missing in a form

Low priority:
- Avoid form submission unless all fields are filled correctly (as in core and space user forms)

Related issues in Sentry:
- [Parameter 'visa_id' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/421/events/816)
- [Parameter 'id_status' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/417)
- [Parameter 'prelevement' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/430)
- [Parameter 'id_quote' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/426)
- [Parameter 'id_eventtype' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/424)
- [Parameter 'ref_protocol' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/419)
- [Parameter 'pricing' is not in the request](https://sentry.genouest.org/sentry/pfm/issues/418)

### PR DESCRIPTION
**Pull request by gaetanHerve**
closes #300

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,9 +2,13 @@
 
 ## 2.2
 
+<<<<<<< HEAD
+* [forms] prevent errors on form submissions
+=======
```
Comment (osallou) [inline CHANGELOG.md:7]: you commit some merge conflicts....

File: Framework/FormAdd.php [hunk 2]
```diff
@@ -90,12 +90,12 @@ public function addLabel($name, $values){
      * @param array $choicesid List of choices Ids
      * @param array $values List of default values
      */
-    public function addSelect($name, $label, $choices, $choicesid, $values = array()) {
+    public function addSelect($name, $label, $choices, $choicesid, $values = array(), $isMandatory=false) {
```
Comment (osallou) [inline Framework/FormAdd.php:93]: there is a addSelectManadatory, why adding new field here?

File: Modules/resources/Controller/ResourcesinfoController.php [hunk 3]
```diff
@@ -460,13 +461,14 @@ public function respsAction($id_space, $id_resource) {
...
+            $_SESSION['flash'] = ResourcesTranslator::StatusNeeded($lang);
+        }
 
-
-        $lang = $this->getLanguage();
         $form = new Form($this->request, "respsform");
         $formAdd = new FormAdd($this->request, "respaddform");
         $formAdd->addSelect("id_user", CoreTranslator::User($lang), $choicesU, $choicesidU, $resps);
-        $formAdd->addSelect("id_status", ResourcesTranslator::Status($lang), $choicesS, $choicesidS, $rstatus);
+        $formAdd->addSelect("id_status", ResourcesTranslator::Status($lang), $choicesS, $choicesidS, $rstatus, isMandatory:true);
```
Comment (osallou) [inline Modules/resources/Controller/ResourcesinfoController.php:471]: why not addselectmanadory like others?

File: Framework/FormAdd.php (hunk 2 above)
Comment (gaetanHerve) [reply Framework/FormAdd.php:93]: there is a Form::addSelectMandatory(), not a FormAdd::addSelectMandatory() :/.
FormAdd is used in forms where  user can add lines (like admins list in space configuration)

**Pull request merged**